osl/interface_linux.go

```diff
@@ -130,6 +131,13 @@ func (n *networkNamespace) Interfaces() []Interface {
 	return ifaces
 }
 
+func (i *nwIface) DisableLearning() bool {
+	i.Lock()
+	i.Unlock()
```

Comment:
Yes, it should have been defer. Thanks.

As mentioned in the description the flooding to the local ports is better avoided. I have pushed a PR with a different approach to address this issue, https://github.com/docker/libnetwork/pull/1792. I will close this PR.